add graphql data provider for data query

BillyWooo opened the pull request:
add new data provider graphql interface.

resolves  #1030 
resolves  #1029
resolves  #1024

Kailai-Wang committed:
adjust docker tag for tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
use integritee/pallets master branch

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
add teeracle to tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Kailai-Wang committed:
switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

Commit:
Merge branch 'tee-dev' of github.com:litentry/litentry-parachain into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
add back the dummy method for testing purpose

Commit:
fix make clippy

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
small cleanup

Kailai-Wang committed:
add back teeracle

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
use specific branch of tee-primitives (todo:revert it)

Kailai-Wang committed:
correctify branch for tee-primitives

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
temporarily use TEE pallets master branch

Kailai-Wang committed:
add EnsureOrigin impl and tests

Kailai-Wang committed:
its better to place it in runtime_common

Kailai-Wang committed:
Merge branch '895-use-real-enclave-signer' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Commit:
remove the dummy extrinsic (used for testing previous)

Commit:
rename ruleset -> assertion

Commit:
update renaming

Commit:
revert balance type

Kailai-Wang committed:
Cherry-Pick: 47386434
Use rococo as dev network (#918)
* add rococo staging
* cargo update
* update README
* fix bad english

Commit:
according to upstream, update pallets branch to 'develop'

Commit:
fix build error

Commit:
add rococo docker file

BillyWooo committed:
Update tee-dev branch into polkadot v0.9.29 (#971)

* merge tee-dev branch into dev branch (#902)

* adjust docker tag for tee-dev

* use integritee/pallets master branch

* add teeracle to tee-dev

* switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

* add back the dummy method for testing purpose

* fix make clippy

* small cleanup

* add back teeracle

* use specific branch of tee-primitives (todo:revert it)

* correctify branch for tee-primitives

* temporarily use TEE pallets master branch

* add EnsureOrigin impl and tests

* its better to place it in runtime_common

* remove the dummy extrinsic (used for testing previous)

Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>

* 0.9.29 update (#891)

* feat: rough update to 0.9.29

* debug:temp fmt

* debug: 0.9.29 without trystate

* debug: Integration with latest try-runtime

* debug: update weight template

* debug: fmt

* debug: patch toml try-runtime

* debug: fmt

* debug: revert template non-weight part

* debug: add teeracle try-runtime

* debug: cargo lock

* fix Cargo.lock (#904)

* lift the bridge transfer restriciton and bump version (#907)

* use hash to compute the raw payload (#910)

* remove Litmus sudo (#876)

* sudo

* change Root into Root +democracy

* debug: recover chainspec rootkey

* fix clippy

* try to fix #900

* add debug log #900

* make CI tests pass after remove litmus `Sudo`

* fmt

* debug

* debug

* rococo/litentry align with litmus

* remove useless test code

Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>

* Use rococo as dev network (#918)

* add rococo staging

* cargo update

* update README

* fix bad english

* Bump aes-gcm from `be0d6c2` to `5eb6d1a` (#925)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `be0d6c2` to `5eb6d1a`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/be0d6c2355aa4dee288e18446e49d3765eaed207...5eb6d1a046371411d062db62671da29799beba7a)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.2.0 to 2.3.0 (#926)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/commits/v2.3.0)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Minor update (#928)

* add dwellir bootnode

* bump litmus runtime version

* cargo update

* feat: relocate remove_sudo migration code (#936)

* feat: relocate remove_sudo migration code

* debug: fmt

* debug: change staking parameter (#931)

* debug: change staking parameter

* debug: bump rococo.litentry spec version

* Use sccache in Dockerfile (#937)

* try to switch to github-runner

* use sccache in Dockerfile for self-hosted runner

* fix syntax

* download sccache

* fix path and use pre-built taplo

* intentionally change lib.rs

* intentionally update a pallet only

* try to add/stop sccache server

* minor update

* update README

* try to use rust-cache

* use /root

* use GITHUB_REF_NAME in hash-key

* I dislike github syntax

* another try

* deprecate set-output

* revert back lib.rs

* dintinguish between release and prod

* fix RUN order

* use head_ref or ref_name

Co-authored-by: BillyWooo <yang@litentry.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: WMQ <46511820+wangminqi@users.noreply.github.com>
Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
add BUILDKIT explicitly

Kailai-Wang committed:
Merge tee worker repo (#987)

* initial adding tee-worker src

* fix link path

* adjust path and content

* fix path

* fmt

* add ignore/exclude path

* fix the path for upload/download artifacts

* fix path

BillyWooo committed:
add graphql data provider for data query

Kailai-Wang commented on tee-worker/litentry/core/data-providers/src/graphql.rs:
```diff
@@ -0,0 +1,186 @@
+	Ethereum,
+}
+pub struct VerifiedCredentialsIsHodlerIn {
+	pub addresses: Vec<String>,
+	pub from_date: String,
+	pub network: VerifiedCredentialsNetwork,
+	pub token_address: String,
+	pub mini_balance: f64,
```
It's a bit confusing that the input is a vector of addresses but only with one single network.

There's an `address` in `IsHodlerOutStruct` too, what is that? 

I think it's helpful if we can attach some exemplary request/response as comments

BillyWooo replied:
Sure. It's a bit confusing. Will add some example/comments.

Kailai-Wang commented on tee-worker/litentry/core/stf-task/receiver/src/stf_task_receiver.rs:
```diff
@@ -108,6 +119,21 @@ where
 				},
+				Assertion::A4 => {
+					let from_date = "2022-10-16T00:00:00Z".to_string();
+					#[cfg(feature = "clock")]
```
where is the `clock` feature used?

BillyWooo replied:
it's for `Utc::now()` within chrono-sgx compile condition.

Kailai-Wang replied:
I mean we never activated this feature in Cargo.toml for `chrono-sgx`? `clock` is not included as default features if I'm not mistaken

Kailai-Wang committed:
Switch to local dependencies (#1037)

* use local dependencies

* fix docker path

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into graphql_data_provider

Kailai-Wang committed:
Try to fix the docker image pushing (#1040)

* debug: try download and push

* add debug commands

* remove file checker

* remove defaults

* add pwd

* make sure to chekcout code

* checkout code

* add CI yml to detect list

Kailai-Wang committed:
remove unneeded CI steps (#1049)

0xverin committed:
tee-worker ts-tests (#1039)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

BillyWooo committed:
add multiconfig for launch workers with local development / staging env / production env

0xverin committed:
Ts tests (#1053)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

BillyWooo committed:
add comment

Kailai-Wang committed:
Add more testcases for IMP mock (#1057)

* add more testcases

* fix compile

* fix file checker in tee-worker-ci

Co-authored-by: ericzhang <jingleizhang@users.noreply.github.com>

BillyWooo committed:
982 renaming issue (#1052)

* rename `link_identity` to `create_identity`

* rename `unlink_identity` to `remove_identity`

* add up the missing ones

* add up missing ones 2

* make fmt happy

* hope not more missed...

* remove not needed file

* update old name from merge (link -> create)

Commit:
Merge branch 'tee-dev' into 1000_muti_config_to_launch_workers

jingleizhang committed:
Merge branch 'tee-dev' into graphql_data_provider

BillyWooo committed:
Merge branch '1000_muti_config_to_launch_workers' of github.com:litentry/litentry-parachain into 1000_muti_config_to_launch_workers

BillyWooo committed:
move hardcode const to input parameter

BillyWooo committed:
make clippy happy

0xverin committed:
link web3 and verify web3 (#1072)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

* add ethers provider&&wallet

* add eth endpoint

* add @ethersproject/providers&&ethers

* substrate&&ethereum tests

* web3 types

* nonce

* change methods name

* change methods name

* change events

* change methods name

* change event

* modify method name

Kailai-Wang committed:
Emit new challenge code for each `create_identity` (#1074)

* some more renaming

* adjustment to IMP mock

* change to sgx-runtime pallets

jingleizhang committed:
Merge branch 'tee-dev' into 1000_muti_config_to_launch_workers

Kailai-Wang committed:
Return encrypted IDGraph when changed (#1080)

* add id_graph to ocall extrinsics

* adjust callback extrinsic

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* slight renaming

* change maxVerificationDelay to 30min

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

* add ethers provider&&wallet

* add eth endpoint

* add @ethersproject/providers&&ethers

* substrate&&ethereum tests

* web3 types

* nonce

* change methods name

* change methods name

* change events

* change methods name

* change event

* modify method name

* ts-test support `id_graph`

* fix bug

* remove log

* resolve conflicts

Co-authored-by: Verin1005 <daqingchong0809@gmail.com>
Co-authored-by: zzz <zhiming.zhong@litentry.com>

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into graphql_data_provider

BillyWooo committed:
Tee worker polkadot v0.9.32 (#1093)

* merge tee-dev branch into dev branch (#902)

* adjust docker tag for tee-dev

* use integritee/pallets master branch

* add teeracle to tee-dev

* switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

* add back the dummy method for testing purpose

* fix make clippy

* small cleanup

* add back teeracle

* use specific branch of tee-primitives (todo:revert it)

* correctify branch for tee-primitives

* temporarily use TEE pallets master branch

* add EnsureOrigin impl and tests

* its better to place it in runtime_common

* remove the dummy extrinsic (used for testing previous)

Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>

* 0.9.29 update (#891)

* feat: rough update to 0.9.29

* debug:temp fmt

* debug: 0.9.29 without trystate

* debug: Integration with latest try-runtime

* debug: update weight template

* debug: fmt

* debug: patch toml try-runtime

* debug: fmt

* debug: revert template non-weight part

* debug: add teeracle try-runtime

* debug: cargo lock

* fix Cargo.lock (#904)

* lift the bridge transfer restriciton and bump version (#907)

* use hash to compute the raw payload (#910)

* remove Litmus sudo (#876)

* sudo

* change Root into Root +democracy

* debug: recover chainspec rootkey

* fix clippy

* try to fix #900

* add debug log #900

* make CI tests pass after remove litmus `Sudo`

* fmt

* debug

* debug

* rococo/litentry align with litmus

* remove useless test code

Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>

* Use rococo as dev network (#918)

* add rococo staging

* cargo update

* update README

* fix bad english

* Bump aes-gcm from `be0d6c2` to `5eb6d1a` (#925)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `be0d6c2` to `5eb6d1a`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/be0d6c2355aa4dee288e18446e49d3765eaed207...5eb6d1a046371411d062db62671da29799beba7a)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.2.0 to 2.3.0 (#926)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/commits/v2.3.0)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Minor update (#928)

* add dwellir bootnode

* bump litmus runtime version

* cargo update

* feat: relocate remove_sudo migration code (#936)

* feat: relocate remove_sudo migration code

* debug: fmt

* debug: change staking parameter (#931)

* debug: change staking parameter

* debug: bump rococo.litentry spec version

* Use sccache in Dockerfile (#937)

* try to switch to github-runner

* use sccache in Dockerfile for self-hosted runner

* fix syntax

* download sccache

* fix path and use pre-built taplo

* intentionally change lib.rs

* intentionally update a pallet only

* try to add/stop sccache server

* minor update

* update README

* try to use rust-cache

* use /root

* use GITHUB_REF_NAME in hash-key

* I dislike github syntax

* another try

* deprecate set-output

* revert back lib.rs

* dintinguish between release and prod

* fix RUN order

* use head_ref or ref_name

Co-authored-by: BillyWooo <yang@litentry.com>

* Update v0.9.30 (#942)

* some deps update to v0.9.30

* deps update to v0.9.30

* format code

* change deps

* fix docker file

* fix docker file

* change docker file

* change docker file

* fix docker yml file

* fix docker yml file

* fix docker yml file

* add install dependencis shell

* fix install dependencis shell

* fix install dependencis shell

* fix docker file

* fix yml file

* add sudo permisson

* add some dep

Co-authored-by: Chengcheng Shi <Chengcheng@litentry>

* optimize cli commands (#947)

* optimize cli commands

* Change some command's names

* format docker file

Co-authored-by: SCC <root@LAPTOP-73QQ3GCK.localdomain>

* Add onfinality script (#952)

* minor update

* bump version and cargo update

* [benchmarking bot] Auto commit generated weights files (#951)

Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>

* try to use new weights

* fix typo

* add missing mods

* fix compile

* add onFinality image version update

* fix indention

* add continue-on-error

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>

* Enable parachain staking extrinsic (#955)

* debug: change extrinsic filter

* debug: fmt and clippy

* debug: revert cargo.lock

* Add release bot GHA (#957)

* add release bot yml

* small update

* Issue 958 (#959)

* fix parachain staking benchmark code

* fix syntax

* omit CollatorSelection benchmark code

* [benchmarking bot] Auto commit generated weights files (#961)

Co-authored-by: Jinsipang <Jinsipang@users.noreply.github.com>

Co-authored-by: Chengcheng Shi
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Bump serde_json from 1.0.87 to 1.0.88 (#967)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.87 to 1.0.88.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.87...v1.0.88)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Optimize parachain-staking benchmark code to shorten running time (#974)

* remove reward for delegator warn logs

* fix benchmark shell

* fix some syntax

* fix shell

* fix shells

* optimized benchmark code and shell

* fix some params

* optimize benchmark code

* [benchmarking bot] Auto commit generated weights files (#973)

Co-authored-by: Jinsipang

* fix benchmark code

* add staking benchmark code at runtime

* fix benchmark code

* fix benchmark code

* fix benchmark code

* format code

* change benchmark code args

Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Bump aes-gcm from `86250ac` to `6ac2d1c` (#970)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `86250ac` to `6ac2d1c`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/86250ac216bb0230c178765303265c6f037d9408...6ac2d1cec6f6658ed5fbb1befa93a370d567f354)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump serde from 1.0.147 to 1.0.148 (#991)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.147 to 1.0.148.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.147...v1.0.148)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump serde_json from 1.0.88 to 1.0.89 (#990)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.88 to 1.0.89.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.88...v1.0.89)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* feat: add PolkadotXCM filter (#993)

* update Polkadot v0.9.32 (#1043)

* update deps to polkaodt-v0.9.31

* update node cli

* fix some test code

* update to polkadot-v0.9.32

* fix buuld benchmark failed and run test failed issues

* fix node code

* [benchmarking bot] Auto commit generated weights files (#1046)

Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>

* Support standalone node for parachain (#1059)

* use unified crate version and cargo update

* initial add standalone mode

* fix chain spec id

* fix clippy

* add script to start standalone node

* add Makefile

* use 9944 as parachain port consistently

* adjust bridge endpoint

* minor update

* recover docker image

* minor update

* bump version

* update README

* Bump decode-uri-component from 0.2.0 to 0.2.2 in /ts-tests (#1060)

Bumps [decode-uri-component](https://github.com/SamVerschueren/decode-uri-component) from 0.2.0 to 0.2.2.
- [Release notes](https://github.com/SamVerschueren/decode-uri-component/releases)
- [Commits](https://github.com/SamVerschueren/decode-uri-component/compare/v0.2.0...v0.2.2)

---
updated-dependencies:
- dependency-name: decode-uri-component
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Backport from release-v0.9.13 banch (#1067)

* minor fix of text search

* fix small typo

* Bump express from 4.17.2 to 4.18.2 in /ts-tests (#1071)

Bumps [express](https://github.com/expressjs/express) from 4.17.2 to 4.18.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.17.2...4.18.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Runtime upgrade simulation (#1081)

* update script

* use relative path

* update script

* improve arg handling

* fix trap

* simulmation runtime upgrade demo impl

* runtime-upgrade ts impl

* fix some syntax

* fix makefile

* optimized scripts && cli

* use master branch of fork-off-substrate

* optimize ts code

* fix description

* fix action sytanx

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* optimize ts-code & cli

* adjustment of script and GHA

* try to fix CI

Co-authored-by: Scc <root@LAPTOP-73QQ3GCK.localdomain>
Co-authored-by: Scc <chengcheng@litentry.com>

* update dependencies from fork pallets to origin integritee pallets v0.9.32 (#1083)

* switch to temp branch

* update to integritee-pallets v0.9.32

* update pallets repo source

* add up missing lock file

* [benchmarking bot] Auto commit generated weights files

* [benchmarking bot] Auto commit generated weights files (#1082)

Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>

* Revert "[benchmarking bot] Auto commit generated weights files (#1082)"

This reverts commit 6b11ff212ce5f047f2172eda10be9748d888570e.

* Revert "Merge remote-tracking branch 'origin/benchmarking-bot-3656108355' into pr-integritee-pallets-v0.9.32"

This reverts commit 3cb2961b5246083a257de0f8d4a17f5325e4fb07, reversing
changes made to d93c2447c12db33dcd86243ce54d58d3484eea6e.

Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: zhizming-zhong <zhiming.zhong@litentry.com>

* Bump serde from 1.0.149 to 1.0.150 (#1085)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.149 to 1.0.150.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.149...v1.0.150)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.3.0 to 2.3.1 (#1086)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/compare/v2.3.0...v2.3.1)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Add test local pallets benchmarking shell (#1087)

* add local benchmark testing shell

* fix benmarking scripts

* fix some syntax

* fix some syntax

* fix some syntax

Co-authored-by: SCC <chengcheng@litentry.com>

* Revert "Add test local pallets benchmarking shell (#1087)" (#1091)

This reverts commit 1c1ee636674925b99cbdd621924cc09bf3445744.

* update tee-worker to upstream/master (3cc02342)

* bump up version to 0.9.32

* cargo build / fmt / clippy / test pass

* parachain build

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: WMQ <46511820+wangminqi@users.noreply.github.com>
Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: SCC <87012234+Jinsipang@users.noreply.github.com>
Co-authored-by: Chengcheng Shi <Chengcheng@litentry>
Co-authored-by: SCC <root@LAPTOP-73QQ3GCK.localdomain>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>
Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>
Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>
Co-authored-by: ericzhang <jingleizhang@users.noreply.github.com>

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into tee-dev

BillyWooo committed:
update tee worker ci

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
refactor terminal parameter into json config file

BillyWooo committed:
update dependency version

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into graphql_data_provider

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into graphql_data_provider

BillyWooo committed:
rm not used files

BillyWooo committed:
Merge branch '1000_muti_config_to_launch_workers' into graphql_data_provider

BillyWooo committed:
cargo update & update global variable

BillyWooo committed:
fix path

BillyWooo committed:
Merge branch '1000_muti_config_to_launch_workers' into graphql_data_provider

BillyWooo committed:
refactor code with new graphql API example

Kailai-Wang commented:
Maybe we need to close and reopen the PR to trigger CI

DogLi commented on tee-worker/litentry/core/assertion-build/src/a4.rs:
```diff
@@ -0,0 +1,99 @@
+
+				return Ok(())
+			}
+		} else {
+			continue
+		}
```
for loop is finished here, no need `continue`. Maybe we need to run `cargo clippy` for this?

BillyWooo replied:
Yes, you are right. Didn't notice to delete it.
And yes. Clippy is not yet applied. Waiting for some small last step update. Maybe I should convert this PR to draft first.

DogLi commented on tee-worker/litentry/core/assertion-build/src/a4_7_12.rs:
```diff
@@ -0,0 +1,134 @@
+) -> Result<()> {
+	let mut client = GraphQLClient::new();
+
+	let mut addresses: Vec<String> = vec![];
+	// let from_date = format!("{:?}", Utc::now());
+	let mut network = VerifiedCredentialsNetwork::Litentry;
+	// let token_address = LIT_TOKEN_ADDRESS.to_string();
+	// let mini_balance = 0f64;
+
+	for identity in identities {
+		if let IdentityWebType::Web3(web3_type) = identity.web_type {
+			match identity.handle {
+				IdentityHandle::Address20(addr) =>
+					addresses.push(from_utf8(&addr).unwrap().to_string()),
+				IdentityHandle::Address32(addr) =>
+					addresses.push(from_utf8(&addr).unwrap().to_string()),
+				IdentityHandle::String(addr) =>
+					addresses.push(from_utf8(&addr).unwrap().to_string()),
+			}
+
+			match web3_type {
+				Web3Network::Substrate(SubstrateNetwork::Polkadot) =>
+					network = VerifiedCredentialsNetwork::Polkadot,
+				Web3Network::Substrate(SubstrateNetwork::Kusama) =>
+					network = VerifiedCredentialsNetwork::Kusama,
+				Web3Network::Substrate(SubstrateNetwork::Litentry) =>
+					network = VerifiedCredentialsNetwork::Litentry,
+				Web3Network::Substrate(SubstrateNetwork::Litmus) =>
+					network = VerifiedCredentialsNetwork::Litmus,
+				Web3Network::Evm(EvmNetwork::Ethereum) =>
+					network = VerifiedCredentialsNetwork::Ethereum,
+				_ =>
+					return Err(Error::Assertion4_7_12Error(
+						"network type not implemented".to_string(),
+					)),
+			}
+		};
+	}
```
I think it's the same in `a4.rs`, we can write a function to get the network and addresses.

BillyWooo replied:
Sorry. This file should be removed later. I've already convert this PR to draft. I am waiting for some decision to adapt the code.

DogLi commented on tee-worker/litentry/core/data-providers/src/graphql.rs:
```diff
@@ -0,0 +1,191 @@
+			.map_err(|e| Error::RequestError(format!("{:?}", e)))?;
+
+		if let Some(value) = response.data.get("data") {
+			let is_hodler_out: IsHodlerOut = serde_json::from_value(value.clone()).unwrap();
```
need to make sure or add some comment that `unwrap()` is much safe to use here, or return an Invalid data error.

BillyWooo committed:
Merge branch 'dev' into graphql_data_provider

BillyWooo committed:
merge assertion4 and assertion12: both query LIT; cleanup code;

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into graphql_data_provider

jingleizhang committed:
Merge branch 'dev' into graphql_data_provider

Kailai-Wang committed:
Merge branch 'dev' into graphql_data_provider

Kailai-Wang committed:
Merge branch 'dev' into graphql_data_provider

Kailai-Wang commented on tee-worker/litentry/core/assertion-build/src/a4.rs:
```diff
@@ -0,0 +1,98 @@
+	GraphQLClient, VerifiedCredentialsIsHodlerIn, VerifiedCredentialsNetwork,
+};
+
+const LIT_TOKEN_ADDRESS: &str = "0xb59490aB09A0f526Cc7305822aC65f2Ab12f9723";
```
We need to add a comment where this address comes from

BillyWooo replied:
add comment

Kailai-Wang commented on tee-worker/litentry/core/data-providers/src/lib.rs:
```diff
@@ -117,6 +125,12 @@ impl DataProvidersStatic {
 	pub fn set_discord_auth_token(&mut self, v: String) {
 		self.discord_auth_token = v;
 	}
+	pub fn set_graphql_url(&mut self, v: String) {
+		self.graphql_url = v;
+	}
+	pub fn set_graphql_auth_key(&mut self, v: String) {
+		self.graphql_auth_key = v;
+	}
```
It seems we lack where these setters are used

BillyWooo replied:
updated.

Kailai-Wang commented on tee-worker/litentry/core/assertion-build/src/a4.rs:
```diff
@@ -0,0 +1,98 @@
+pub fn build(
+	identities: BoundedVec<Identity, MaxIdentityLength>,
+	from_date: String,
+	mini_balance: f64,
```
do you mean `min_balance`?

BillyWooo replied:
updated. min_balance

Kailai-Wang commented on tee-worker/litentry/core/data-providers/src/graphql.rs:
```diff
@@ -0,0 +1,191 @@
+		GraphQLClient { client }
+	}
+
+	pub fn verified_credentials_is_hodler(
```
It might be a bit confusing that this `fn` and the members in `IsHodlerOut` have the same name

BillyWooo replied:
changed.

Kailai-Wang commented on tee-worker/litentry/core/assertion-build/src/a7.rs:
```diff
@@ -0,0 +1,78 @@
+				addresses,
+				from_date: from_date.clone(),
+				network: VerifiedCredentialsNetwork::Polkadot,
+				token_address: String::from(""),
```
empty token address?

BillyWooo replied:
Yes. Based on the TDF-graphql API, to query DOT doesn't need token-address.

Kailai-Wang commented on tee-worker/litentry/core/assertion-build/src/a4.rs:
```diff
@@ -0,0 +1,98 @@
+				mini_balance,
+			};
+			let is_hodler_out = client.verified_credentials_is_hodler(credentials);
+			if let Ok(_hodler_out) = is_hodler_out {
```
If I understand it correctly, `addresses` can only have max one address, so if we have `n` linked identities for LIT, we will generate `n` VCs? It seems incorrect

BillyWooo replied:
Not generate `n` VCs. Once one hodler_out matches, it will return. Line 93.

Kailai-Wang replied:
I see, so the logic is if **any** of the linked identities is hodler, we'll have a VC saying `is_hodler` is true. And what if all of them are false? We still need to generate a VC.

It's fine that we leave it blank for now, but we desperately need a test soon :(

BillyWooo replied:
`We still need to generate a VC.` then we need to add an `else` branch for this `if let ...`.

Kailai-Wang replied:
I think it should be inside it, `if let` only checks if the output is meaningful but doesn't check if it's holder. Anyway it's another topic

Kailai-Wang reviewed (commented):
I have raised a few questions.

Are the older review comments resolved? If so, we should mark them as "resolved"

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into graphql_data_provider

BillyWooo committed:
update based on PR comments

Kailai-Wang reviewed (approved):
Thanks! 

Let's not forget we'd need multiple tests once the VC part is done
